notebook2/Makefile

```diff
@@ -32,8 +33,9 @@ build/images/%/_PUSHED_LATEST_TAG: build/images/%/_BUILT
 
 $(IMAGE_SHORT_NAMES): %: build/images/%/_BUILT
 
+
 notebook-worker-images: $(PUSHED)
-	cat $^ > $@
+	cat $^ > $@ && test $(APISERVER) && echo $(APISERVER) >> $@
```

Comment:
Why are you appending the api server name to this file?

Reply:
The apiserver worker image, and building logic lives in the ../apiserver. I had spoken to Cotton about whether to continue managing those built images in that directory, or move that to notebook/notebook2. He seemed to prefer the former, and asked me to call out to read the image, using "existing patterns".

I wasn't entirely certain what the pattern was, but searching through other projects, I found fairly regular use of a pattern involving reading metadata from a running kube pod. Most notably, `apiserver/Makefile` uses this as well

```
run-hail-jupyter-pod: HAIL_JUPYTER_IMAGE=$(shell kubectl get deployment apiserver -o jsonpath='{.metadata.annotations.hail-jupyter-image}')
```

This guarantees, like the existing notebook/notebook2 worker-images pattern, that only pushed images may be specified, and also ensures that the hail-jupyter image that apiserver is currently running in cluster with is used. If apiserver is not deployed, notebook2 will entirely fail to build, which seems the desired behavior, since it would otherwise depend on an impossible-to-use image.

However, happy to use a different/better pattern if I misunderstood.

cc @cseed

Reply:
I didn't read the metadata closely enough. I thought it was the server name, not the notebook image.

Reply:
Actually, why is the variable called API_SERVER then? Maybe can we go through this in person on Monday so I understand this part of the system better?

Reply:
It's the Jupyter image built by and to be used with the apiserver.  APISERVER_JUPYTER_IMAGE would be more specific.